app/routes/events/components/EventList.js

```diff
@@ -1,157 +1,172 @@
-  return (
-    <div
-      style={{ borderColor: colorForEvent(event.eventType) }}
-      className={styles.eventItem}
-    >
-      <div>
-        <Link to={`/events/${event.id}`}>
-          <h3 className={styles.eventItemTitle}>{event.title}</h3>
-          {event.totalCapacity > 0 && (
-            <Attendance
-              registrationCount={event.registrationCount}
-              totalCapacity={event.totalCapacity}
-              style={{ marginLeft: '5px', color: 'black' }}
-            />
-          )}
-        </Link>
-
-        <div className={styles.eventTime}>
-          <Time time={event.startTime} format="ll HH:mm" />
-          {` • ${event.location}`}
-        </div>
-        {showTags && (
-          <Flex wrap className={styles.tagList}>
-            {event.tags.map((tag, index) => (
-              <Tag key={index} tag={tag} small />
-            ))}
-          </Flex>
-        )}
-      </div>
 
-      <div className={styles.companyLogo}>
-        {event.cover && <Image src={event.cover} />}
-      </div>
-    </div>
-  );
-}
-
-type EventListGroupProps = {
+const EventListGroup = ({
+  name,
+  field = 'startTime',
+  events = []
+}: {
   name: string,
-  events: Array</*TODO: Event */ any>
-};
-
-function EventListGroup({ name, events = [] }: EventListGroupProps) {
+  field: 'activationTime' | 'startTime',
+  events: Array<Event>
+}) => {
   return isEmpty(events) ? null : (
     <div className={styles.eventGroup}>
       <h2 className={styles.heading}>{name}</h2>
       {events.map((event, i) => (
-        <EventItem key={i} event={event} />
+        <EventItem key={i} event={event} field={field} />
       ))}
     </div>
   );
-}
+};
 
 type EventListProps = {
-  events: Array<any>,
-  actionGrant: /* TODO: ActionGrant */ any,
-  icalToken: /* TODO: IcalToken */ string,
+  events: Array<Event>,
+  actionGrant: ActionGrant,
+  icalToken: IcalToken,
   showFetchMore: boolean,
   fetchMore: () => Promise<*>
 };
 
-// $FlowFixMe
-const EventList = (props: EventListProps) => {
-  const events = groupEvents(props.events);
-  const { icalToken, showFetchMore, fetchMore } = props;
-  return (
-    <div className={styles.root}>
-      <Helmet title="Arrangementer" />
-      <Toolbar actionGrant={props.actionGrant} />
-      <EventListGroup name="Denne uken" events={events.currentWeek} />
-
-      <EventListGroup name="Neste uke" events={events.nextWeek} />
-
-      <EventListGroup name="Senere" events={events.later} />
-
-      {isEmpty(props.events) && (
-        <EmptyState icon="book-outline" size={40}>
-          <h2 className={styles.noEvents}>Ingen kommende arrangementer</h2>
-        </EmptyState>
-      )}
-      {showFetchMore && <Button onClick={fetchMore}>Last inn mer</Button>}
-      <div className={styles.bottomBorder} />
-      <EventFooter icalToken={icalToken} />
-    </div>
-  );
+type Option = {
+  filterFunc: Event => boolean,
+  label: string,
+  field: 'activationTime' | 'startTime'
```

Comment:
You can use https://flow.org/en/docs/types/utilities/#toc-keys to fix this, or just make a custom type `type FieldType = "activationTime" | "startTime"